test/client/doc.js

```diff
@@ -14,28 +14,39 @@ describe('client query subscribe', function() {
     expect(doc).equal(doc2);
   });
 
-  it('calling doc.destroy unregisters it', function() {
-    var doc = this.connection.get('dogs', 'fido');
-    expect(this.connection.getExisting('dogs', 'fido')).equal(doc);
+  it('calling doc.destroy unregisters it', function(done) {
+    var connection = this.connection;
+    var doc = connection.get('dogs', 'fido');
+    expect(connection.getExisting('dogs', 'fido')).equal(doc);
 
-    doc.destroy();
-    expect(this.connection.getExisting('dogs', 'fido')).equal(undefined);
+    doc.destroy(function(err) {
+      if (err) return done(err);
+      expect(connection.getExisting('dogs', 'fido')).equal(undefined);
 
-    var doc2 = this.connection.get('dogs', 'fido');
-    expect(doc).not.equal(doc2);
+      var doc2 = connection.get('dogs', 'fido');
+      expect(doc).not.equal(doc2);
+      done();
+    });
+
+    // destroy is async
+    expect(connection.getExisting('dogs', 'fido')).equal(doc);
   });
 
-  it('getting then destroying then getting returns a new doc object', function() {
-    var doc = this.connection.get('dogs', 'fido');
-    doc.destroy();
-    var doc2 = this.connection.get('dogs', 'fido');
-    expect(doc).not.equal(doc2);
-    expect(doc).eql(doc2);
+  it('getting then destroying then getting returns a new doc object', function(done) {
+    var connection = this.connection;
+    var doc = connection.get('dogs', 'fido');
+    doc.destroy(function(err) {
+      if (err) return done(err);
+      var doc2 = connection.get('dogs', 'fido');
+      expect(doc).not.equal(doc2);
+      expect(doc).eql(doc2);
+      done();
+    });
   });
 
-  it('doc.destroy() calls back', function(done) {
+  it('doc.destroy() works without a callback', function() {
```

Comment:
Maybe keep both of theses cases?

```js
it('doc.destroy() calls back', ...);
it('doc.destroy() works without a callback', ...);
```

The case where it does invoke the callback is a good thing to have covered by tests IMO.

Reply:
@curran the previous 2 tests use `destroy` with a callback, so this use case is covered sufficiently IMHO.

Reply:
Aha yes you're right. Apologies for missing that.

Reply:
Aha yes you're right. Apologies for missing that.

In that case I'd say this looks good to merge.